### PR TITLE
Update CSV output format

### PR DESCRIPTION
**Pull request by jeanbez**
Update CSV output format by including headers, adding missing columns for some rows, and standardizing the labels.

### CODE REVIEW

File: h5bench_patterns/h5bench_append.c [hunk 1]
```diff
@@ -679,16 +679,17 @@ main(int argc, char *argv[])
...
-            fprintf(params.csv_fs, "Raw modify rate, %.3f, GB/sec\n", raw_rate_gbs);
-            fprintf(params.csv_fs, "Observed modify rate, %.3f, GB/sec\n", or_gbs);
+            fprintf(params.csv_fs, "metric, value, unit\n");
+            fprintf(params.csv_fs, "operation, %s, %s\n", "append", "");
+            fprintf(params.csv_fs, "ranks, %d, %s\n", NUM_RANKS, "");
+            fprintf(params.csv_fs, "total compute time, %llu, %s\n", total_sleep_time_us / (1000 * 1000),
+                    "seconds");
+            fprintf(params.csv_fs, "total size, %lu, %s\n", total_size_gb, "GB");
+            fprintf(params.csv_fs, "raw time, %.3f, %s\n", rrt_s, "seconds");
+            fprintf(params.csv_fs, "raw rate, %.3f, %s\n", raw_rate_gbs, "GB/s");
```
Comment (sbyna) [inline h5bench_patterns/h5bench_append.c:689]: If the data size is too small, GB/s values may be zeros. Should there be MB/s and GB/s based on the size of the data?

File: h5bench_patterns/h5bench_overwrite.c [hunk 2]
```diff
@@ -656,16 +656,17 @@ main(int argc, char *argv[])
...
-            fprintf(params.csv_fs, "Raw modify rate, %.3f, GB/sec\n", raw_rate_gbs);
-            fprintf(params.csv_fs, "Observed modify rate, %.3f, GB/sec\n", or_gbs);
+            fprintf(params.csv_fs, "metric, value, unit\n");
+            fprintf(params.csv_fs, "operation, %s, %s\n", "overwrite", "");
+            fprintf(params.csv_fs, "ranks, %d, %s\n", NUM_RANKS, "");
+            fprintf(params.csv_fs, "total compute time, %llu, %s\n", total_sleep_time_us / (1000 * 1000),
+                    "seconds");
+            fprintf(params.csv_fs, "total size, %lu, %s\n", total_size_gb, "GB");
+            fprintf(params.csv_fs, "raw time, %.3f, %s\n", rrt_s, "seconds");
+            fprintf(params.csv_fs, "raw rate, %.3f, %s\n", raw_rate_gbs, "GB/s");
```
Comment (sbyna) [inline h5bench_patterns/h5bench_overwrite.c:666]: Same as the above comments.

File: h5bench_patterns/h5bench_read.c [hunk 3]
```diff
@@ -549,16 +549,17 @@ main(int argc, char *argv[])
...
-            fprintf(params.csv_fs, "Raw read rate, %.3f, GB/sec\n", raw_rate_gbs);
-            fprintf(params.csv_fs, "Observed read rate, %.3f, GB/sec\n", or_gbs);
+            fprintf(params.csv_fs, "metric, value, unit\n");
+            fprintf(params.csv_fs, "operation, %s, %s\n", "read", "");
+            fprintf(params.csv_fs, "ranks, %d, %s\n", NUM_RANKS, "");
+            fprintf(params.csv_fs, "total compute time, %llu, %s\n", total_sleep_time_us / (1000 * 1000),
+                    "seconds");
+            fprintf(params.csv_fs, "total size, %lu, %s\n", total_size_gb, "GB");
+            fprintf(params.csv_fs, "raw time, %.3f, %s\n", rrt_s, "seconds");
+            fprintf(params.csv_fs, "raw rate, %.3f, %s\n", raw_rate_gbs, "GB/s");
```
Comment (sbyna) [inline h5bench_patterns/h5bench_read.c:559]: Similar to previous comments about small data sizes.

File: h5bench_patterns/h5bench_write.c [hunk 4]
```diff
@@ -1132,23 +1132,19 @@ main(int argc, char *argv[])
...
+            fprintf(params.csv_fs, "metric, value, unit\n");
+            fprintf(params.csv_fs, "operation, %s, %s\n", "write", "");
+            fprintf(params.csv_fs, "ranks, %d, %s\n", NUM_RANKS, "");
+            fprintf(params.csv_fs, "collective data, %s, %s\n", params.data_coll == 1 ? "YES" : "NO", "");
+            fprintf(params.csv_fs, "collective meta, %s, %s\n", params.meta_coll == 1 ? "YES" : "NO", "");
+            fprintf(params.csv_fs, "total compute time, %llu, %s\n", total_sleep_time_us / (1000 * 1000),
+                    "seconds");
+            fprintf(params.csv_fs, "total size, %lu, %s\n", total_size_gb, "GB");
+            fprintf(params.csv_fs, "raw time, %.3f, %s\n", rwt_s, "seconds");
+            fprintf(params.csv_fs, "raw rate, %.3f, %s\n", raw_rate_gbs, "GB/s");
```
Comment (sbyna) [inline h5bench_patterns/h5bench_write.c:1144]: Similar to previous comments about small data sizes.

File: h5bench_patterns/h5bench_append.c (hunk 1 above)
Comment (jeanbez) [reply h5bench_patterns/h5bench_append.c]: Done!

File: h5bench_patterns/h5bench_append.c [hunk 5]
```diff
@@ -679,16 +679,17 @@ main(int argc, char *argv[])
...
+            fprintf(params.csv_fs, "metadata time, %.3f, %s\n", meta_time_s, "seconds");
+            fprintf(params.csv_fs, "observed rate, %.3f, %s\n", or_gbs, "GB/sec");
```
[the remark this answers is not in the document]
Comment (jeanbez) [reply h5bench_patterns/h5bench_append.c]: Done!

File: h5bench_patterns/h5bench_overwrite.c (hunk 2 above)
Comment (jeanbez) [reply h5bench_patterns/h5bench_overwrite.c]: Done!

File: h5bench_patterns/h5bench_read.c (hunk 3 above)
Comment (jeanbez) [reply h5bench_patterns/h5bench_read.c]: Done!

File: h5bench_patterns/h5bench_write.c (hunk 4 above)
Comment (jeanbez) [reply h5bench_patterns/h5bench_write.c]: Done!

File: h5bench_patterns/h5bench_write_unlimited.c [hunk 6]
```diff
@@ -1101,23 +1101,19 @@ main(int argc, char *argv[])
...
+            fprintf(params.csv_fs, "metric, value, unit\n");
+            fprintf(params.csv_fs, "operation, %s, %s\n", "write unlimited", "");
+            fprintf(params.csv_fs, "ranks, %d, %s\n", NUM_RANKS, "");
+            fprintf(params.csv_fs, "collective data, %s, %s\n", params.data_coll == 1 ? "YES" : "NO", "");
+            fprintf(params.csv_fs, "collective meta, %s, %s\n", params.meta_coll == 1 ? "YES" : "NO", "");
+            fprintf(params.csv_fs, "total compute time, %llu, %s\n", total_sleep_time_us / (1000 * 1000),
+                    "seconds");
+            fprintf(params.csv_fs, "total size, %lu, %s\n", total_size_gb, "GB");
+            fprintf(params.csv_fs, "raw time, %.3f, %s\n", rwt_s, "seconds");
+            fprintf(params.csv_fs, "raw rate, %.3f, %s\n", raw_rate_gbs, "GB/s");
```
[the remark this answers is not in the document]
Comment (jeanbez) [reply h5bench_patterns/h5bench_write_unlimited.c]: Done!

**Review by sbyna — APPROVED** [review]
Looks good.

**Pull request merged**